### DISCUSSION TITLE
Managing Product Features

### DISCUSSION THREAD

**Discussion by adam-code-labx** [lunarphp/lunar#1020] [Feature Requests]
I would say perhaps not everyone will use this feature depend on the shop and the data they have, this could be behind a feature-flag to enable after install. In which case you could argue this maybe better suited as an add-on, however as you guys are still in beta I would say behind feature-flag would be the best option.

Anyway to explain this in a little more detail.
As you know I am currently migrating one of my clients away from Prestashop. So of course we need to retain their existing data.
This the way prestashop manage features https://doc.prestashop.com/display/PS17/Managing+Product+Features

I would say real world example for you to see the features and values would be shein.co.uk
https://www.shein.co.uk/Lettuce-Edge-Leopard-Mesh-Crop-Top-p-1493253-cat-1738.html
Open description panel to see more.

Color:	Multicolor
Style:	Sexy
Pattern Type:	Leopard Print
...

The idea will be to define a global list of features and their values, then later which we will discuss in another issue associate features by product type and of course override from the product section in the hub.

For example if we were to setup a new product using the above example "Women's T-Shirts"
Then from the product edit page we would see a list of all the groups then all the the staff of the shop would need todo is choose the value(s). 

Another real world example would be ebay take a random item for example https://www.ebay.co.uk/itm/123818164938
scroll down to Item specifics you will see there is a list of features and their values for Photo Canvas. take another item this time a car https://www.ebay.co.uk/itm/155074745651 you will see here a completely different set of features. Some of which allow to display multiple values such as In-Car Audio

**Comment by adam-code-labx** [comment]
I have already started to implement this for my client and started a PR #405
Currently put into draft to allow time for some improvements refactor etc. 
Plus should give you guys a chance to review 👍 

<img width="1911" alt="Screenshot 2022-07-15 at 11 19 27" src="https://user-images.githubusercontent.com/53559175/179313379-747dcaf1-cc2b-4d50-994c-34b531c435f5.png">

**Comment by glennjacobs** [comment]
Thanks for taking the time to explain in detail. We'll take a look as a team on Monday all being well.

**Comment by adam-code-labx** [comment]
Hi @glennjacobs sorry it's been way too hot to work today and the forecast is the same for tomorrow so trying to get a few hours work done during my night shift. As my garden office is like a sauna in the day 😂

I noticed when renaming my branch which I did so that I could separate out the features from the attribute options it actually auto closed the PR. I am guessing from my side I will have to start another PR for the updated branch?
Unless you're able to reopen from you side?

Seems like Github restricts reopening PR's if you rename branches which is certainly annoying 😭

**Comment by glennjacobs** [comment]
Weirdly, it won't allow me to reopen it either. Guess you'll have to make a new one.

**Comment by glennjacobs** [comment]
I've not had a chance to review with @alecritson yet, as we had server issues yesterday and he's now on annual leave. 

There's a chance we may suggest doing this as an addon, as our intention was always to try keep the core to functionality that everyone is likely to use.

**Comment by adam-code-labx** [comment]
Good morning @glennjacobs 
To be honest it is a feature everyone should use, if they know it's here I am sure they will 😁

I will create another PR and put this behind a feature-flag for now which I will include in the new PR.
Also will record my screen once implemented with your demo site so you can see this in action.
Then perhaps we can start a poll via discord and see what the other users think before deciding to convert into an addon.

Hope to get finished on the other product settings PR's this week then move on to some refactoring suggestions.
With @alecritson being on annual leave are you still going to be able to actively review my contributions yourself or with the rest of your team during the next couple of weeks?

**Comment by glennjacobs** [comment]
Alec is back tomorrow, so should be fine for this week.

**Comment by adam-code-labx** [comment]
Oh ok thought you meant off on summer holidays 😂

**Comment by glennjacobs** [comment]
Hi @adam-code-labx 

Again, I could do with an understanding of what problem this solves. Currently, we have product attributes which from my knowledge appear to offer the same functionality.

**Comment by adam-code-labx** [comment]
@glennjacobs I will demonstrate to you later on how this is used as I can not publicly share my client info which is why I provided you with the shein and ebay examples.
